xhttp/xhttpserver/tls.go

```diff
@@ -250,20 +249,3 @@ func NewTlsConfig(t *Tls, extra ...PeerVerifier) (*tls.Config, error) {
 	tc.BuildNameToCertificate() // nolint: staticcheck
 	return tc, nil
 }
-
-type connectionStateKey struct{}
-
-// ConnectionState returns the tls.ConnectionState from the given context.
-func ConnectionState(ctx context.Context) (cs tls.ConnectionState, present bool) {
-	cs, present = ctx.Value(connectionStateKey{}).(tls.ConnectionState)
-	return
-}
-
-// SetConnectionState associates a tls.ConnectionState with the given context.
-func SetConnectionState(ctx context.Context, cs tls.ConnectionState) context.Context {
-	return context.WithValue(
-		ctx,
-		connectionStateKey{},
-		cs,
-	)
-}
```

Comment:
removing since `http.Server.ConnContext` can't be used as previously planned

Reply:
Good find!  Thanks.